src/lib/integration_test_lib/wait_condition.ml

```diff
@@ -132,8 +141,20 @@ struct
           | Payment { source_pk; receiver_pk; amount = paid_amt; token_id = _ }
             when Public_key.Compressed.equal source_pk sender_pub_key
                  && Public_key.Compressed.equal receiver_pk receiver_pub_key
-                 && Currency.Amount.equal paid_amt amount ->
-              true
+                 && Currency.Amount.equal paid_amt amount -> (
+              match command_type with Send_payment -> true | _ -> false )
+          | Stake_delegation dl -> (
+              match dl with
+              | Set_delegate
```

Comment:
it surprises me that this Set_delegate's record doesn't have a field for "amount", does that matter?

Reply:
That's normal. You can't choose an amount of stake to delegate -- you can only delegate your entire wallet's balance.